src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.cs

```diff
@@ -388,7 +388,33 @@ private bool MethodCallOrArrayIndex(AcceptedCharacters acceptedCharacters)
                     }
                     return MethodCallOrArrayIndex(acceptedCharacters);
                 }
-                if (CurrentSymbol.Type == CSharpSymbolType.Dot)
+                if (At(CSharpSymbolType.QuestionMark))
+                {
+                    var next = Lookahead(count: 1);
+
+                    if (next != null)
+                    {
+                        if (next.Type == CSharpSymbolType.Dot)
+                        {
+                            // Accept null propagation dot operator (?.).
+                            AcceptAndMoveNext();
+                            AcceptAndMoveNext();
+
+                            // If the next piece after the ?. is a keyword or identifier then we want to continue.
+                            return At(CSharpSymbolType.Identifier) || At(CSharpSymbolType.Keyword);
+                        }
+                        else if (next.Type == CSharpSymbolType.LeftParenthesis ||
+                            next.Type == CSharpSymbolType.LeftBracket)
+                        {
+                            // It's either ?[ or ?(, accept the question mark because it's a null propagation operator.
```

Comment:
https://github.com/dotnet/roslyn/wiki/New-Language-Features-in-C%23-6#null-conditional-operators specifically says
> Note that an invocation (a parenthesized argument list) cannot immediately follow the ? operator – that would lead to too many syntactic ambiguities.

The workaround for delegates is a null-conditional `Invoke` call e.g. `delegate?.Invoke(...)`